src/ast/types/blocks-and-statements.ts

```diff
@@ -46,16 +46,26 @@ export interface EnhancedForStatement {
   kind: "EnhancedForStatement";
 }
 
-export type StatementWithoutTrailingSubstatement = Block | ExpressionStatement | DoStatement;
+export type StatementWithoutTrailingSubstatement = Block | ExpressionStatement | DoStatement | ReturnStatement;
 
-export type ExpressionStatement = MethodInvocation | Assignment;
+export interface ExpressionStatement {
+  kind: "ExpressionStatement";
+  stmtExp: StatementExpression
+}
+
+export type StatementExpression = MethodInvocation | Assignment;
```

Comment:
Since `ExpressionStatement ::= StatementExpression`, I think keeping `ExpressionStatement = MethodInvocation | Assignment` can avoid unnecessary indirection.

Reply:
ECE requires `ExpressionStatement` to be a `Node`.